listen/property-changes.js

```diff
@@ -205,51 +253,33 @@ function dispatchEach(listeners, key, value) {
     }
 }
 
-PropertyChanges.prototype.dispatchBeforeOwnPropertyChange = function (key, listener) {
+PropertyChangesP.dispatchBeforeOwnPropertyChange = function (key, listener) {
     return PropertyChanges.dispatchOwnPropertyChange(this, key, listener, true);
 };
 
-PropertyChanges.prototype.makePropertyObservable = function (key) {
+//PropertyChangesP.__state__ = null;
+PropertyChangesP.__overriddenPropertyDescriptors__ = null;
+PropertyChangesP.makePropertyObservable = function (key) {
     // arrays are special.  we do not support direct setting of properties
     // on an array.  instead, call .set(index, value).  this is observable.
     // 'length' property is observable for all mutating methods because
     // our overrides explicitly dispatch that change.
-    if (Array.isArray(this)) {
-        return;
-    }
+	//console.timeStamp("makePropertyObservable "+key);
+	//if("__state__" in this && this.__state__ && key in this.__state__) return;
+
+	//Benoit: What's the point of this?
+	//if(this.__state__ && key in this.__state__) return;
```

Comment:
Note that making a property change observable needs only be done once, regardless of how many times observers are installed on that property. Doing so repeatedly would create an ever growing, never shrinking chain of thunks.